rasa/core/policies/memoization.py

```diff
@@ -241,11 +239,11 @@ def predict_action_probabilities(
 
         return result
 
-    def persist(self, path: Text) -> None:
+    def persist(self, path: Union[Text, Path]) -> None:
```

Comment:
optional: how about defining template methods for `memorized_file` and `data` to de-duplicate the actual saving / loading. That's obviously a larger refactoring so potentially out of the scope for this PR.

Reply:
I had this in my initial implementation, but realised that would be inconsistent with some of the other policies which have very custom `persist()` methods (for example the SklearnPolicy)

Reply:
since Sklearn was the only case deviating from the default pattern, I switched to a template approach. ready for another review